pandas-stubs/core/indexes/datetimes.pyi

```diff
@@ -90,7 +93,7 @@ def date_range(
     start: str | DatetimeLike | None = ...,
     end: str | DatetimeLike | None = ...,
     periods: int | None = ...,
-    freq: str | BaseOffset = ...,
+    freq: str | timedelta | BaseOffset = ...,
```

Comment:
Can you also add `pd.Timedelta` as a valid argument here?  It works as well (and for `bdate_range`)